stout/do-all.h

```diff
@@ -0,0 +1,172 @@
+#include <atomic>
+#include <tuple>
+#include <variant>
+
+#include "stout/compose.h"
+#include "stout/terminal.h"
+
+namespace stout {
+namespace eventuals {
+namespace detail {
+
+struct _DoAll {
+  template <typename K_, typename... Eventuals_>
+  struct Adaptor {
+    Adaptor(K_& k)
+      : k_(k) {}
+    Adaptor(Adaptor&& that)
+      : k_(that.k_),
+        values_(that.values_),
+        counter_(that.counter_.load(std::memory_order_relaxed)) {
+    }
```

Comment:
`that.values_` has the following type:
std::tuple<std::variant<...>>
That's why `that.values_.empty()` is illegal, since `std::tuple` doen't have empty() method.
Should I use `std::tuple_size_v` instead?